EIPS/eip-XXXX.md

```diff
@@ -0,0 +1,91 @@
+---
+eip: XXX
+title: eth/70 - partial block receipt lists
+description: Adds a facility for paginating block receipts in the p2p protocol
+author: Felix Lange <fjl@ethereum.org>, Jochem Brouwer (@jochem-brouwer), Giulio Rebuffo (@Giulio2002)
+discussions-to:
+status: Final
+type: Standards Track
+category: Networking
+created: 2025-06-16
+requires: 7642, 7825
+---
+
+## Abstract
+
+This EIP modifies the 'eth' p2p protocol to allow requesting partial block receipt lists.
+
+## Motivation
+
+As Ethereum moves toward a higher block gas limit on mainnet, the worst-case total size of
+a block receipts list also becomes larger, and may eventually exceed the 10MB message size
+limit commonly applied in clients. This can lead to sync failures.
+
+## Specification
+
+Modify the encoding for receipts in the `Receipts (0x10)` message as follows:
+
+- (eth/69): `[request-id: P, [[receipt₁, receipt₂], ...]]`
+
+- (eth/70): `[request-id: P, lastBlockIncomplete: {0,1}, [[receipt₁, receipt₂], ...]]`
```

Comment:
Do we need to spec behavior if the flag is set incorrectly? If it sends all receipts but marks it as "partial" I think we are still fine (?) and do not have to disconnect. But this would then send a request which would thus yield 0 receipts for the first block... :thinking:

Reply:
I don't think we need to document this case in the EIP. Either way I think it's OK to define it more thoroughly in the spec later. For the EIP, it is more important to document the basic idea.